lib/registry.dart

```diff
@@ -260,17 +260,17 @@ class Registry<T> {
       // Response is even-length list of (id, element) pairs.
       var cache = _cache;
       var count = response.length ~/ 2;
-      var result = List<T>.filled(count, null);
+      var result = List<T?>.filled(count, null);
       for (var i = 0; i < count; i++) {
         var id = response[i * 2] as int;
-        var element = response[i * 2 + 1] as T;
-        element = cache.register(id, element);
+        T? element = response[i * 2 + 1] as T;
+        element = cache.register(id, element) as T?;
         result[i] = element;
       }
       return result;
```

Comment:
This looks wrong. It returns a `List<T?>` and the completer needs a `List<T>`.

The callback should probably be ` (List<T> response) { ... }` to capture that.
(It worked before because the result *was* a `List<T>`, but it won't with a `List<T?>`).